algoliasearch-common/src/main/java/com/algolia/search/inputs/analytics/ABTest.java

```diff
@@ -0,0 +1,38 @@
+package com.algolia.search.inputs.analytics;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+public class ABTest {
+  // Properties available at construction time
+  public String name;
```

Comment:
All the POJOs are beans, so for consistency, put all fields private and generate getters/setters if needeed.
For example: https://github.com/algolia/algoliasearch-client-java-2/blob/master/algoliasearch-common/src/main/java/com/algolia/search/responses/FacetHit.java

Reply:
Yes, I get the consistency reason and after reading again few things, it feels really wrong to use public fields. Thanks for the review about that. I'll remove all the public fields and just add getters (apparently, you don't need both getter and setter with Jackson, because getter=serialization+deserialization whereas setter=deserialization only, still according to http://www.baeldung.com/jackson-field-serializable-deserializable-or-not).

For the record, I think it's the opposite (all JavaBeans are POJOs but not all POJOs are JavaBeans) according to what I found on internet around this subject. My brain right now:
![giphy](https://user-images.githubusercontent.com/3115191/42282776-e30a895e-7fa7-11e8-857a-255171603493.gif)